pkg/querier/querier.go

```diff
@@ -72,8 +74,17 @@ func (cfg *Config) Validate() error {
 	return nil
 }
 
+// Querier can select logs and samples and handle query requests.
+type Querier interface {
+	logql.Querier
+	Label(ctx context.Context, req *logproto.LabelRequest) (*logproto.LabelResponse, error)
+	Query(params logql.Params) logql.Query
+	Series(ctx context.Context, req *logproto.SeriesRequest) (*logproto.SeriesResponse, error)
+	Tail(ctx context.Context, req *logproto.TailRequest) (*Tailer, error)
```

Comment:
I was hoping it would be enough to only implement `logql.Querier`. However, we might want to support Label and Series as well. Not sure about tail.